### PR TITLE
refactor(multipooler): introduce ruleStore for shard rule management

### PR DESCRIPTION
**Pull request by dweitzman**
Replace the insertHistoryRecord approach with a typed ruleStore
abstraction that owns the current_rule and rule_history tables.

- New ruleStorer interface (observePosition, updateRule) enables test
  isolation via fakeRuleStore, replacing direct DB mocks in manager tests
- ruleUpdateBuilder provides a builder API for constructing rule updates
  with optional fields (leader, cohort, WAL position, operation, etc.)
- buildNodePosition consolidates row-to-proto parsing shared by both the
  read (observePosition) and write (updateRule) paths
- newPoolerID/toPoolerIDs now return approximate values alongside errors,
  letting best-effort callers (e.g. BeginTerm revoke) use the result
  while strict callers (e.g. validateStandbyIDs) check the error
- Schema operations move from pg_multischema into rule_store where they
  logically belong

### CODE REVIEW

File: go/common/pgprotocol/client/extended.go [hunk 1]
```diff
@@ -354,11 +355,57 @@ func argToParam(arg any) ([]byte, error) {
 	case time.Time:
 		// Use RFC3339 format which PostgreSQL understands.
 		return []byte(v.Format(time.RFC3339Nano)), nil
+	case []string:
```
Comment (dweitzman) [inline go/common/pgprotocol/client/extended.go:363]: @GuptaManan100, added you as a reviewer because this PR implements support for `TEXT[]` type in postgres

File: go/services/multipooler/manager/rule_store.go
Comment (dweitzman) [inline go/services/multipooler/manager/rule_store.go:1]: This file is the meat of the PR. Everything else is in support of the changes here

File: go/services/multipooler/manager/rule_store.go [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+func (pm *MultiPoolerManager) createCurrentRuleTable(ctx context.Context) error {
+	execCtx, cancel := context.WithTimeout(ctx, 500*time.Millisecond)
+	defer cancel()
+	if err := pm.exec(execCtx, `CREATE TABLE IF NOT EXISTS multigres.current_rule (
+		shard_id                  BYTEA PRIMARY KEY,
+		coordinator_term          BIGINT NOT NULL,
+		rule_subterm              BIGINT NOT NULL,
+		leader_id                 TEXT,
+		coordinator_id            TEXT,
+		cohort_members            TEXT[] NOT NULL,
```
Comment (dweitzman) [inline go/services/multipooler/manager/rule_store.go:166]: This `TEXT[]` for cohort is why I made a change to the protocol & executor

File: go/services/multipooler/manager/pg_replication.go [hunk 3]
```diff
@@ -64,32 +64,49 @@ type poolerID struct {
...
 // - SetPrimaryConnInfo: standby's application_name when connecting to primary
 // - ConfigureSynchronousReplication: standby names in synchronous_standby_names
+//
+// On validation failure an approximate poolerID is returned alongside the error.
+// The approximate appName is "{cell}_{name}" with missing fields replaced by
+// "<unknown>". For overlong names the full (invalid) string is returned as-is.
+// Callers that require a strictly valid appName must check the error. Callers that
+// can tolerate an approximate name (e.g. for logging or informational responses)
+// may use the returned value regardless.
 func newPoolerID(id *clustermetadatapb.ID) (poolerID, error) {
```
Comment (dweitzman) [inline go/services/multipooler/manager/pg_replication.go:74]: This might be a little wacky, but it's potentially temporary: some RPCs are running the postgres application name instead of the topoclient `ID`, but it's theoretically possible to have an ID that can't be turned into a valid application name (we should block pooler startup in that case, ideally)

The hack here is to say that we return a conversion error but also return a placeholder non-empty value so that we can express the cohort for a rule where some of the members are invalid application names

This will be less of an issue later on after we migrate the multiorch to deal with rules instead of application names directly. Then we won't necessarily need to serialize bad IDs into application names anymore, we'll just reject requests to take action on a pooler if the rule is invalid

File: go/services/multipooler/manager/rpc_manager_test.go [hunk 4]
```diff
@@ -932,18 +922,19 @@ func TestPromote_WithElectionMetadata(t *testing.T) {
 	pm.mu.Lock()
 	assert.Equal(t, clustermetadatapb.PoolerType_PRIMARY, pm.multipooler.Type)
 	pm.mu.Unlock()
-	assert.NoError(t, mockQueryService.ExpectationsWereMet())
 
-	// Note: We don't directly test that insertLeadershipHistory is called with the correct metadata
-	// because that would require mocking the database layer. The leadership history functionality
-	// will be tested by the actual implementation. This test verifies that the Promote method
-	// accepts the new parameters without error and completes successfully.
+	update := fakeRules.assertPromoteRecorded(t)
```
Comment (dweitzman) [inline go/services/multipooler/manager/rpc_manager_test.go:926]: Unit tests have the option to use a fake rule store now instead of mocking SQL directly. The SQL for rule changes is complicated and returns quite a few columns, so this will hopefully be easier for unit tests to manage

File: go/common/pgprotocol/client/extended.go [hunk 5]
```diff
@@ -354,11 +360,57 @@ func argToParam(arg any) ([]byte, error) {
...
+	if s == "" {
+		return true
+	}
+	for _, c := range s {
+		if c == '{' || c == '}' || c == ',' || c == '"' || c == '\\' || c == ' ' || c == '\t' || c == '\n' {
+			return true
+		}
+	}
+	return false
+}
```
Comment (mkindahl) [inline go/common/pgprotocol/client/extended.go]: I wonder if this is enough. PostgreSQL strings can contain control characters and general unicode code points, so just checking for the "obvious" ASCII characters would not be sufficient. See my comment above.

File: go/common/pgprotocol/client/extended.go [hunk 6]
```diff
@@ -354,11 +360,57 @@ func argToParam(arg any) ([]byte, error) {
...
+	if len(elems) == 0 {
+		return "{}"
+	}
+	var b strings.Builder
+	b.WriteByte('{')
+	for i, e := range elems {
+		if i > 0 {
+			b.WriteByte(',')
+		}
+		if needsQuoting(e) {
```
Comment (mkindahl) [inline go/common/pgprotocol/client/extended.go]: There is the function `strings.ContainsAny`, but I do not think this logic is sufficient to figure out if quoting is needed since the `e` can be any unicode character, not just ASCII.

```suggestion
		if e == "" || strings.ContainsAny(e, `{},"\ `+"\t\n")
```

File: go/common/pgprotocol/client/extended.go [hunk 7]
```diff
@@ -354,11 +360,57 @@ func argToParam(arg any) ([]byte, error) {
...
+		return []byte(encodeStringArray(v)), nil
 	default:
 		return nil, fmt.Errorf("unsupported type: %T", arg)
 	}
 }
 
+// encodeStringArray encodes a []string as a PostgreSQL text-format array literal (e.g. {foo,bar}).
+// Elements that require quoting (containing braces, commas, backslashes, double-quotes, or whitespace)
+// are enclosed in double-quotes with internal double-quotes and backslashes escaped.
+func encodeStringArray(elems []string) string {
```
Comment (mkindahl) [inline go/common/pgprotocol/client/extended.go]: I don't think this works correctly if you pass in arbitrary UTF-8 encoded unicode. I think that you do not need to check for the quote characters as you do below, you just need to check for backslash and quote and just copy the remaining bytes. You can also always quote the value, which makes the logic simpler.

File: go/common/pgprotocol/client/extended_test.go [hunk 8]
```diff
@@ -414,3 +414,39 @@ func TestWriteBindWithNullParams(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, []byte("hello"), p3)
 }
+
+func TestEncodeStringArray(t *testing.T) {
```
Comment (mkindahl) [inline go/common/pgprotocol/client/extended_test.go:418]: Missing tests for unicode characters, for example:

```go
t.Run("slice with non-ascii runes", func(t *testing.T) {
		assert.Equal(t, `{"ᚼᛅᛁᛚ","ᚼᛅᛁᛗᚱ"}`, encodeStringArray([]string{"ᚼᛅᛁᛚ", "ᚼᛅᛁᛗᚱ"}))
})
```

File: go/common/sqltypes/arrays.go [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+	for i < len(inner) {
+		c := inner[i]
+		switch {
+		case c == '"' && !quoted:
+			quoted = true
+		case c == '"' && quoted:
+			quoted = false
+		case c == '\\' && quoted && i+1 < len(inner):
+			i++
+			cur.WriteByte(inner[i])
```
Comment (mkindahl) [inline go/common/sqltypes/arrays.go]: What if `c == '\\'` but the next character does _not_ exist? Shouldn't that generate an error?

File: go/common/sqltypes/arrays_test.go [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+		result, err := ParseTextArray(`{"foo,bar","baz"}`)
+		require.NoError(t, err)
+		assert.Equal(t, []string{"foo,bar", "baz"}, result)
+	})
+
+	t.Run("quoted element with backslash escape", func(t *testing.T) {
+		result, err := ParseTextArray(`{"foo\"bar"}`)
+		require.NoError(t, err)
+		assert.Equal(t, []string{`foo"bar`}, result)
+	})
```
Comment (mkindahl) [inline go/common/sqltypes/arrays_test.go:61]: Missing error test cases for strings like `{"foo\`, `{"foo\\,}`, and `{"foo","bar"`.

Also correct string cases that should work are `{foo, bar}` (note space).

File: go/services/multipooler/manager/manager.go [hunk 11]
```diff
@@ -108,6 +108,7 @@ type MultiPoolerManager struct {
 	consensusState  *ConsensusState
 	topoLoaded      bool
 	consensusLoaded bool
+	rules           ruleStorer
```
Comment (mkindahl) [inline go/services/multipooler/manager/manager.go:111]: Typo? Shouldn't it be `ruleStore`?

File: go/services/multipooler/manager/pg_replication_test.go [hunk 12]
```diff
@@ -51,56 +51,70 @@ func TestNewPoolerID(t *testing.T) {
...
+			id:              &clustermetadatapb.ID{Cell: "us_west", Name: "replica-1"},
+			expectError:     true,
+			expectedAppName: "us_west_replica-1",
+		},
+		{
+			name:            "name contains underscore",
+			id:              &clustermetadatapb.ID{Cell: "zone1", Name: "replica_1"},
+			expectError:     true,
+			expectedAppName: "zone1_replica_1",
+		},
```
Comment (mkindahl) [inline go/services/multipooler/manager/pg_replication_test.go:112]: Are names ever "deconstructed" to get the `Cell` and the `Name`? If so, could it be that using an _ in the cell or the name could cause confusion as to what it part of the cell and what is part of the name?

So, for example, the `{Cell: "us_west", Name: "1_replica"}` will be indistinguishable from `{Cell: "us_west_1", Name: "replica"}`.

File: go/services/multipooler/manager/rule_store.go [hunk 13]
```diff
@@ -0,0 +1,709 @@
...
+	"github.com/multigres/multigres/go/common/timeouts"
+	"github.com/multigres/multigres/go/common/topoclient"
+	clustermetadatapb "github.com/multigres/multigres/go/pb/clustermetadata"
+	mtrpcpb "github.com/multigres/multigres/go/pb/mtrpc"
+	"github.com/multigres/multigres/go/services/multipooler/executor"
+)
+
+// ruleStorer is the interface for reading and writing the current shard rule.
+// *ruleStore implements this; tests use fakeRuleStore.
+type ruleStorer interface {
```
Comment (mkindahl) [inline go/services/multipooler/manager/rule_store.go:33]: The name is very close to the concrete structure `ruleStore` so there is going to be mixups. Maybe be very specific and call it `ruleStoreInterface`?

File: go/services/multipooler/manager/rule_store.go [hunk 14]
```diff
@@ -0,0 +1,709 @@
...
+
+func newRuleUpdate(termNumber int64, coordinatorID *clustermetadatapb.ID, eventType, reason string, createdAt time.Time) *ruleUpdateBuilder {
+	return &ruleUpdateBuilder{
+		termNumber:    termNumber,
+		coordinatorID: coordinatorID,
+		eventType:     eventType,
+		reason:        reason,
+		createdAt:     createdAt,
+	}
+}
```
Comment (mkindahl) [inline go/services/multipooler/manager/rule_store.go:104]: The normal case is probably to modify an existing rule, so you might want to have an interface where you can do something like

```go
rule.AsBuilder().withLeader(primary).create()
```

File: go/services/multipooler/manager/rule_store.go [hunk 15]
```diff
@@ -0,0 +1,709 @@
...
+		previousTerm = &update.previousRule.coordinatorTerm
+		previousSubterm = &update.previousRule.ruleSubterm
+	}
+
+	// Use the remote operation timeout for history writes. This write validates that synchronous
+	// replication is functioning - it must wait long enough for standbys to connect and acknowledge.
+	execCtx, cancel := context.WithTimeout(ctx, timeouts.RemoteOperationTimeout)
+	defer cancel()
+
+	result, err := rs.queryService.QueryArgs(execCtx, `
```
Comment (mkindahl) [inline go/services/multipooler/manager/rule_store.go:364]: The query is difficult to review since it is so long. I have to trust you that it does the right thing. Offhand I make the following observations:

1. Using `SELECT FOR UPDATE` and then adding one offers the same guarantees as using a sequence and inserting rows into a table, but without the potential gaps in the sequence. Is it important to not have gaps in the sequence of coordinate terms?
2. There is a `SELECT FOR UPDATE` which means that you expect to have multiple sessions running the same statement, but AFAICT, Multipooler would not run multiple such updates at the same time, so why is that `SELECT FOR UPDATE` needed?
3. The `locked` table is using the conditions `coordinator_term = ... AND rule_subterm = ...`, which means that if an error creeps into the logic or the table, this will return zero rows. How will that work with the query? What are the consequences of such a bug?
4. Everything you do in `next_sub` can be added to `locked` as instead, but this is mostly a matter of style.
5. The `shard_id` is always zero. Do we need it then? Can it ever be anything else, or is this just added to deal with shards in the future?

File: go/services/multipooler/manager/rule_store_test.go
Comment (mkindahl) [inline go/services/multipooler/manager/rule_store_test.go:1]: I miss tests. For example, since you use `SELECT FOR UPDATE` it means that you expect conflicting updates, but there is nothing like that here.

**Review by mkindahl — COMMENTED** [review]
Here are a few comments. The SQL statement for doing a rule update looks very complicated, so I wonder if it can be simplified.

We discussed having a single table and just append to the table instead. What are your thoughts around that?

File: go/common/pgprotocol/client/extended_test.go (hunk 8 above)
Comment (dweitzman) [reply go/common/pgprotocol/client/extended_test.go:418]: Added unicode test

File: go/common/pgprotocol/client/extended.go (hunk 7 above)
Comment (dweitzman) [reply go/common/pgprotocol/client/extended.go]: Changed to always quote

File: go/common/sqltypes/arrays_test.go (hunk 10 above)
Comment (dweitzman) [reply go/common/sqltypes/arrays_test.go:61]: Added a bunch of test cases

File: go/common/pgprotocol/client/extended.go (hunk 6 above)
Comment (dweitzman) [reply go/common/pgprotocol/client/extended.go]: Changed to always quote, so this is dead code

File: go/common/pgprotocol/client/extended.go (hunk 5 above)
Comment (dweitzman) [reply go/common/pgprotocol/client/extended.go]: Removed this code in favor of always quoting

File: go/common/sqltypes/arrays.go (hunk 9 above)
Comment (dweitzman) [reply go/common/sqltypes/arrays.go]: Added a test case

File: go/services/multipooler/manager/manager.go (hunk 11 above)
Comment (dweitzman) [reply go/services/multipooler/manager/manager.go:111]: This is the interface name, not the struct name. I don't have strong feelings on interface names, but an `-er` suffix seems consistent with Go conventions: https://go.dev/doc/effective_go#interface-names

File: go/services/multipooler/manager/pg_replication_test.go (hunk 12 above)
Comment (dweitzman) [reply go/services/multipooler/manager/pg_replication_test.go:112]: I think the answer is that we sometimes maybe parse application names back to IDs, but as the refactoring proceeds that will become increasingly unnecessary to the point where it stops happening

My eventual goal is that we generally deal with the proto `ID` type and only convert it to an application name immediately before changing the GUC in postgres

When we get to that point, I can remove this code

In the meantime, I don't think anything that's supposed to work will break because of this change, I just want to avoid returning an empty cohort list just because orch is expecting information in the legacy appName format instead of IDs

File: go/services/multipooler/manager/rule_store.go (hunk 13 above)
Comment (dweitzman) [reply go/services/multipooler/manager/rule_store.go:33]: The `-er` naming is an attempt to follow Go conventions. I'm not personally that attached to Go conventions, but my default thinking is to try to follow them where practical

File: go/services/multipooler/manager/rule_store.go (hunk 14 above)
Comment (dweitzman) [reply go/services/multipooler/manager/rule_store.go:104]: I imagine something like `AsBuilder()` appearing in the future, but for this PR I wanted to minimize unused code since there's plenty interesting happening here already

File: go/services/multipooler/manager/rule_store.go (hunk 15 above)
Comment (dweitzman) [reply go/services/multipooler/manager/rule_store.go:364]: For now I feel most comfortable with `SELECT FOR UPDATE` on a single rules table because I'm not sure how to implement compare-and-swap without it and I predict that we'll want compare-and-swap (and put it into the builder interface).

**Review by dweitzman — COMMENTED** [review]
I've also added a test that runs the SQL query against a real postgres instance

File: go/common/pgprotocol/client/extended_test.go [hunk 16]
```diff
@@ -414,3 +414,47 @@ func TestWriteBindWithNullParams(t *testing.T) {
...
+		assert.Equal(t, `{"zone1_pooler-1","zone1_pooler-2","zone1_pooler-3"}`, encoded)
+	})
+
+	t.Run("unicode elements are encoded correctly", func(t *testing.T) {
+		assert.Equal(t, `{"ᚼᛅᛁᛚ","ᚼᛅᛁᛗᚱ"}`, encodeStringArray([]string{"ᚼᛅᛁᛚ", "ᚼᛅᛁᛗᚱ"}))
+	})
+
+	t.Run("null string is quoted to avoid SQL NULL interpretation", func(t *testing.T) {
+		assert.Equal(t, `{"NULL"}`, encodeStringArray([]string{"NULL"}))
+	})
```
Comment (mkindahl) [inline go/common/pgprotocol/client/extended_test.go:459]: That should not be possible since the extended protocol is used (parameters are passed directly as strings with no SQL-level parsing between), but it does not harm to test this.

File: go/common/sqltypes/arrays_test.go [hunk 17]
```diff
@@ -0,0 +1,227 @@
...
+	t.Run("stray closing brace inside element", func(t *testing.T) {
+		_, err := ParseTextArray(`{foo}bar}`)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "unexpected '}'")
+	})
+
+	t.Run("multi-dimensional array", func(t *testing.T) {
+		_, err := ParseTextArray(`{{a,b},{c,d}}`)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "multi-dimensional")
```
Comment (mkindahl) [inline go/common/sqltypes/arrays_test.go:225]: Since a `\` will read one more rune after it would be good to test that a `\` at the end of the string does not cause problems.

File: go/services/multipooler/manager/rule_store.go [hunk 18]
```diff
@@ -0,0 +1,713 @@
...
+// current_rule holds a single row per shard representing the current cluster rule.
+// It is used as a locking target (SELECT FOR UPDATE) to serialise concurrent
+// writes; rule_history provides the append-only audit log.
+//
+// coordinator_term=0 in the sentinel row means no rule has been applied yet.
+func (rs *ruleStore) createRuleTables(ctx context.Context) error {
+	execCtx, cancel := context.WithTimeout(ctx, 500*time.Millisecond)
+	defer cancel()
+
+	if _, err := rs.queryService.Query(execCtx, `CREATE TABLE IF NOT EXISTS multigres.current_rule (
```
Comment (mkindahl) [inline go/services/multipooler/manager/rule_store.go:160]: This looks dangerous.

If you conditionally create the table, it might have a different definition compared to what this code expects (for example, as a result of a new version of the code). I think you should either get an error proper (and remove the `IF NOT EXISTS`) or drop the table before you create it if this is acceptable.

File: go/services/multipooler/manager/rule_store.go [hunk 19]
```diff
@@ -0,0 +1,713 @@
...
+		created_at                TIMESTAMPTZ NOT NULL
+	)`); err != nil {
+		return mterrors.Wrap(err, "failed to create current_rule table")
+	}
+
+	if _, err := rs.queryService.QueryArgs(execCtx, `
+		INSERT INTO multigres.current_rule
+		  (shard_id, coordinator_term, rule_subterm, cohort_members, created_at)
+		VALUES ($1, 0, 0, '{}', now())
+		ON CONFLICT (shard_id) DO NOTHING`,
```
Comment (mkindahl) [inline go/services/multipooler/manager/rule_store.go:180]: If the definition of the table does not match what you have above, this will fail despite using `ON CONFLICT ... DO NOTHING`. If you create the table above (by removing the `IF NOT EXISTS`), then the conflict resolution is not needed (but it does not hurt either).

File: go/services/multipooler/manager/rule_store.go [hunk 20]
```diff
@@ -0,0 +1,713 @@
...
+		[]byte("0")); err != nil {
+		return mterrors.Wrap(err, "failed to initialize current_rule")
+	}
+
+	// Each row records a cluster state change (promotion, cohort membership, durability policy).
+	// The composite primary key (coordinator_term, rule_subterm) uniquely identifies each rule;
+	// rule_subterm is assigned by the application as MAX(rule_subterm)+1 within a coordinator_term.
+	if _, err := rs.queryService.Query(execCtx, `CREATE TABLE IF NOT EXISTS multigres.rule_history (
```
Comment (mkindahl) [inline go/services/multipooler/manager/rule_store.go:188]: Same as above. If the table already exists, it could have a different definition compared to what the code expects, so this could cause problems. Better get a hard error here if the table should not exist, or drop the table if it is ok to lose the data in it.

File: go/common/sqltypes/arrays.go (hunk 9 above)
Comment (mkindahl) [reply go/common/sqltypes/arrays.go]: I can't see a test case that parses a string that ends with a `\`. Did I miss it?

File: go/services/multipooler/manager/manager.go (hunk 11 above)
Comment (mkindahl) [reply go/services/multipooler/manager/manager.go:111]: Either one works for me. Just wondered if it was intentional.

File: go/services/multipooler/manager/rule_store.go (hunk 15 above)
Comment (mkindahl) [reply go/services/multipooler/manager/rule_store.go:364]: So you have multiple connections updating the row at the same time and you need to serialize these?

**Review by mkindahl — APPROVED** [review]
I think it makes more sense to remove the `IF NOT EXISTS` for the tables, or drop the tables prior to creating them, or we will run into weird errors later if the table already exists.

Approving since I do not think it will cause any issues currently, but probably something we should fix before releasing.

File: go/services/multipooler/manager/rule_store.go [hunk 21]
```diff
@@ -0,0 +1,713 @@
...
+// concurrent writes at the database level in addition to the caller's action lock.
+//
+// Returns the node's position (rule + WAL LSN) at the time of the write,
+// or nil if force mode skipped the write.
+//
+// This operation uses the remote-operation-timeout and will fail if it cannot
+// complete within that time. A timeout typically indicates that synchronous
+// replication is not functioning.
+func (rs *ruleStore) updateRule(ctx context.Context, update *ruleUpdateBuilder) (*clustermetadatapb.NodePosition, error) {
+	if update.force {
```
Comment (dweitzman) [inline go/services/multipooler/manager/rule_store.go:309]: In a follow-up PR I'm going to remove this

File: go/services/multipooler/manager/rule_store.go [hunk 22]
```diff
@@ -0,0 +1,713 @@
...
+		  locked AS (
+		    -- FOR UPDATE serializes concurrent writes at the database level, complementing
+		    -- the action lock held by the caller.
+		    -- Returns zero rows (causing a no-op write and error return) when any condition fails.
+		    -- next_sub: 0 when starting a new coordinator term, otherwise increment within term.
+		    SELECT current_rule.coordinator_term, current_rule.rule_subterm,
+		           current_rule.leader_id, current_rule.cohort_members,
+		           CASE WHEN params.coordinator_term > current_rule.coordinator_term THEN 0
+		                ELSE current_rule.rule_subterm + 1
+		           END AS next_sub
```
Comment (dweitzman) [inline go/services/multipooler/manager/rule_store.go]: I've renamed rule_subterm to leader_subterm and renamed next_sub to next_leader_subterm

File: go/services/multipooler/manager/rule_store.go (hunk 20 above)
Comment (dweitzman) [reply go/services/multipooler/manager/rule_store.go:188]: I think this is a reasonable change to make, but we should make it for all the other multischema DDLs as well. I'll send a follow-up PR

File: go/common/sqltypes/arrays_test.go (hunk 17 above)
Comment (dweitzman) [reply go/common/sqltypes/arrays_test.go:225]: Done
